src/mapping/config/MappingConfiguration.cpp

```diff
@@ -494,6 +494,14 @@ MappingConfiguration::ConfiguredMapping MappingConfiguration::createMapping(
                 "Mesh \"{0}\" was not found while creating a mapping. "
                 "Please correct the to=\"{0}\" attribute.",
                 toMeshName);
+
+  // Check for compatible mesh dimensions
+  PRECICE_CHECK(fromMesh->getDimensions() == toMesh->getDimensions(),
+                "Mapping between meshes of different dimensions is not allowed. "
+                "Please set the same dimensions attribute to meshes \"{}\" and \"{}\", "
+                "or choose different meshes.",
+                fromMeshName, toMeshName);
```

Comment:
@ezonta when adding the geometric multiscale mapping, we will need to wrap this check in an if-clause that checks the mapping type. Something like this:

```c++
if (type == TYPE_GEOMETRIC_MULTISCALE) {
  // ... types of geometric multiscale, allowed dimensions combinations
} else {
  // this check
}
```